tests/test_cli.py

```diff
@@ -77,16 +77,34 @@ def test_unset_non_existent_value(cli, dotenv_file):
         ("auto", "HELLO", "HELLO WORLD", 'HELLO="HELLO WORLD"\n'),
     )
 )
-def test_set_options(cli, dotenv_file, quote_mode, variable, value, expected):
+def test_set_quote_options(cli, dotenv_file, quote_mode, variable, value, expected):
     result = cli.invoke(
         dotenv_cli,
-        ["--file", dotenv_file, "--quote", quote_mode, "set", variable, value]
+        ["--file", dotenv_file, "--export", "false", "--quote", quote_mode, "set", variable, value]
     )
 
     assert (result.exit_code, result.output) == (0, "{}={}\n".format(variable, value))
     assert open(dotenv_file, "r").read() == expected
 
 
+@pytest.mark.parametrize(
+    "dotenv_file,export_mode,variable,value,expected",
+    (
+        (".nx_file.sh", "true", "HELLO", "WORLD", "export HELLO=\"WORLD\"\n"),
+        (".nx_file", "false", "HELLO", "WORLD", "HELLO=\"WORLD\"\n"),
+    )
+)
+def test_set_export(cli, dotenv_file, export_mode, variable, value, expected):
+    result = cli.invoke(
+        dotenv_cli,
+        ["--file", dotenv_file, "--quote", "always", "--export", export_mode, "set", variable, value]
+    )
+    assert result.exit_code == 0, result.output
+
+    assert (result.exit_code, result.output) == (0, "{}={}\n".format(variable, value))
```

Comment:
`result.exit_code` seems to be checked twice here.  Perhaps this could be simplified to make the test clearer?

Reply:
Certainly.